### PR TITLE
support qemu lazy mode

### PR DESCRIPTION
**Pull request by YaoZengzeng**
I make qemu match LazyDriverContext and add three items in QemuContext.So we can launch Qemu with net and block device mounted .The "interface" and "image" in QemuContext is used to store the net and block device configuration.The "callbacks" is used to send the event that devices is ready, so we can start pod.
  Unfortunately, there seems a little problem, at the very begining, we may receive a tty or init message with impossible big length message.So, I do a little trick to find and  read the entire wrong message, and start reading again.Then every thing is OK.But I dont know if it is relevant with lazy mode?

### CODE REVIEW

File: hypervisor/init_comm.go [hunk 1]
```diff
@@ -79,6 +79,11 @@ func ReadVmMessage(conn *net.UnixConn) (*DecodedMessage, error) {
 		res = append(res, buf[:nr]...)
 		read = read + nr
 
+		// if condition satisified, message length may wrong , read again
+		if nr < want && read < needRead {
+			return ReadVmMessage(conn)
+		}
+
```
Comment (laijs) [inline hypervisor/init_comm.go]: why these code is needed? the loop will continue read the compose the whole message?

Comment (YaoZengzeng) [inline hypervisor/init_comm.go]: Because it sometimes will get a impossible long message as follows:
I0426 14:22:34.827787   26408 init_comm.go:91] data length is 582570580
I0426 14:22:34.827794   26408 init_comm.go:72] trying to read 512 bytes
I0426 14:22:34.827804   26408 init_comm.go:72] trying to read 8 bytes
I0426 14:22:34.905799   26408 init_comm.go:87] read 8/8 [length = 0]
I0426 14:22:34.905833   26408 init_comm.go:91] data length is 1462868
I0426 14:22:34.905839   26408 init_comm.go:72] trying to read 512 bytes
I0426 14:22:34.905849   26408 init_comm.go:72] trying to read 8 bytes
Especially when you stop the hyper daemon unproperly ,restart it and launch the first pod.So in order not to be blocked here, I do this to clarify the bad message and start to read a new message.

File: hypervisor/qemu/qemu.go [hunk 2]
```diff
@@ -354,4 +373,68 @@ func (qc *QemuContext) arguments(ctx *hypervisor.VmContext) []string {
 		"-fsdev", fmt.Sprintf("local,id=virtio9p,path=%s,security_model=none", ctx.ShareDir),
 		"-device", fmt.Sprintf("virtio-9p-pci,fsdev=virtio9p,mount_tag=%s", hypervisor.ShareDirTag),
 	)
+
+	//lazy add nic device
+	for i, info := range qc.interfaces {
+		params = append(params,
+		"-netdev", fmt.Sprintf("tap,fd=%d,id=%s", info.fd, info.deviceName),
```
Comment (laijs) [inline hypervisor/qemu/qemu.go]: qemu is launched via daemon mode, the all the fds will be closed. how these code works?

Comment (YaoZengzeng) [inline hypervisor/qemu/qemu.go]: Yes，you're right, in this branch I forget something and didn't test the nic after the qemu launch.In order to fix this , we should modify the code under path /runv/lib/utils/daemon.go:
	//Close all open file descriptors
//	for (i = 0; i < num; i++) {
//		if (fds[i] == 0 || fds[i] == 1 || fds[i] == 2)
//			continue;
//		printf("\n\nI didnt close any file descriptor\n\n");
//		close(fds[i]);
//	}
just don't close the fds.If want to optimize , we can just leave the nic fd open.

File: hypervisor/init_comm.go (hunk 1 above)
Comment (laijs) [inline hypervisor/init_comm.go]: could you split it into another pr?

Comment (laijs) [inline hypervisor/init_comm.go]: it it hard to recover the channel is the previous hyper daemon exited improperly.
we need to find a better solution.

Comment (YaoZengzeng) [inline hypervisor/init_comm.go]: That means I delete this tricky part and just pr the main lazy qemu code???And should I add the code that not closing all fds?

Comment (laijs) [inline hypervisor/init_comm.go]: I will add some code to avoid close some the fds.

Comment (YaoZengzeng) [inline hypervisor/init_comm.go]: OK，I'll make a new PR immediately

Comment (laijs) [inline hypervisor/init_comm.go]: Actually, you don't need to create a new pr for qemu lazy, just use this one.
just create a new pr for ReadVmMessage().
